src/sentry/interfaces/exception.py

```diff
@@ -23,6 +23,797 @@
+        6: 'CLD_CONTINUED',  # [XSI] stopped child has continued
+    },
+
+    # Codes for SIGPOLL
+    7: {
+        1: 'POLL_IN',  # [XSR] Data input available
+        2: 'POLL_OUT',  # [XSR] Output buffers available
+        3: 'POLL_MSG',  # [XSR] Input message available
+        4: 'POLL_ERR',  # [XSR] I/O error
+        5: 'POLL_PRI',  # [XSR] High priority input available
+        6: 'POLL_HUP',  # [XSR] Device disconnected
+    },
+}
+
+# Mach exception codes used in Darwin.
+WELL_KNOWN_MACH_EXCEPTIONS = {
+    1: 'EXC_BAD_ACCESS',  # Could not access memory
+    2: 'EXC_BAD_INSTRUCTION',  # Instruction failed
+    3: 'EXC_ARITHMETIC',  # Arithmetic exception
+    4: 'EXC_EMULATION',  # Emulation instruction
+    5: 'EXC_SOFTWARE',  # Software generated exception
+    6: 'EXC_BREAKPOINT',  # Trace, breakpoint, etc.
+    7: 'EXC_SYSCALL',  # System calls.
+    8: 'EXC_MACH_SYSCALL',  # Mach system calls.
+    9: 'EXC_RPC_ALERT',  # RPC alert
+    10: 'EXC_CRASH',  # Abnormal process exit
+    11: 'EXC_RESOURCE',  # Hit resource consumption limit
+    12: 'EXC_GUARD',  # Violated guarded resource protections
+    13: 'EXC_CORPSE_NOTIFY',  # Abnormal process exited to corpse state
+}
+
+
+def normalize_mechanism_errno(errno, sdk):
+    if not sdk:
+        return
+
+    if 'name' not in errno:
+        errnos = WELL_KNOWN_ERRNO.get(sdk, {})
+        name = errnos.get(errno['number'])
+        if name:
+            errno['name'] = name
+
+
+def normalize_mechanism_signal(signal, sdk):
+    if not sdk:
+        return
+
+    if 'name' not in signal:
+        signals = WELL_KNOWN_SIGNALS.get(sdk, {})
+        name = signals.get(signal['number'])
+        if name:
+            signal['name'] = name
+
+    if sdk != 'darwin':
+        return
+
+    if 'code' in signal and 'code_name' not in signal:
+        codes = WELL_KNOWN_SIGNAL_CODES.get(signal['number'], {})
+        code_name = codes.get(signal['code'])
+        if code_name:
+            signal['code_name'] = code_name
+
+
+def normalize_mechanism_mach_exception(mach):
+    if 'name' not in mach:
+        name = WELL_KNOWN_MACH_EXCEPTIONS.get(mach['exception'])
+        if name:
+            mach['name'] = name
+
+
+def normalize_mechanism_meta(mechanism, sdk_info=None):
+    if mechanism is None or 'meta' not in mechanism:
+        return
+
+    meta = mechanism['meta']
+
+    sdk_name = sdk_info['sdk_name'].lower() if sdk_info else ''
+    if sdk_name in ('ios', 'watchos', 'tvos', 'macos'):
+        sdk = 'darwin'
+    elif sdk_name in ('linux', 'android'):
+        sdk = 'linux'
+    elif sdk_name in ('windows',):
+        sdk = 'windows'
+    else:
+        sdk = None
+
+    if 'errno' in meta:
+        normalize_mechanism_errno(meta['errno'], sdk)
+
+    if 'signal' in meta:
+        normalize_mechanism_signal(meta['signal'], sdk)
+
+    if 'mach_exception' in meta:
+        normalize_mechanism_mach_exception(meta['mach_exception'])
+
+
+def upgrade_legacy_mechanism(data):
```

Comment:
`upgrade_legacy_mechanism` converts from the old mechanism structure used in sentry-cocoa to the new one. It is used in at the very top of `Mechanism.to_python`, as well as in `applecrashreport.py` - the only two places where we parse and re-serialize exception mechanisms.